keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/artifact/ImageHandler.kt

```diff
@@ -32,66 +31,63 @@ class ImageHandler(
   private val baseImageCache: BaseImageCache,
   private val igorService: ArtifactService,
   private val imageService: ImageService,
-  private val diffFingerprintRepository: DiffFingerprintRepository,
   private val publisher: ApplicationEventPublisher,
   private val taskLauncher: TaskLauncher,
   private val defaultCredentials: BakeCredentials
 ) : ArtifactHandler {
 
   override suspend fun handle(artifact: DeliveryArtifact) {
     if (artifact is DebianArtifact) {
-      val latestArtifactVersion = try {
+      val desiredAppVersion = try {
         artifact.findLatestArtifactVersion()
       } catch (e: NoKnownArtifactVersions) {
         log.debug(e.message)
         return
       }
 
-      if (taskLauncher.correlatedTasksRunning(artifact.correlationId(latestArtifactVersion))) {
+      if (taskLauncher.correlatedTasksRunning(artifact.correlationId(desiredAppVersion))) {
         publisher.publishEvent(
           ArtifactCheckSkipped(artifact.type, artifact.name, "ActuationInProgress")
         )
       } else {
-        val latestBaseAmiName = artifact.findLatestBaseAmiName()
-
-        val desired = Image(
-          baseAmiName = latestBaseAmiName,
-          appVersion = latestArtifactVersion,
-          regions = artifact.vmOptions.regions
-        )
-        val current = artifact.findLatestAmi()
-        val diff = DefaultResourceDiff(desired, current)
-
-        if (current != null && diff.isRegionsOnly()) {
-          if (current.regions.containsAll(desired.regions)) {
-            log.debug("Image for ${current.appVersion} contains more regions than we need, which is fine")
-          } else {
-            log.warn("Detected a diff in the regions for ${current.appVersion}: ${diff.toDebug()}")
-            publisher.publishEvent(ImageRegionMismatchDetected(current, artifact.vmOptions.regions))
+        val desiredBaseAmiName = artifact.findLatestBaseAmiName()
+
+        val images = artifact.findLatestAmi(desiredAppVersion)
+        val olderBaseImages = images.filterValues { it.baseImageName != desiredBaseAmiName }
+        val missingRegions = artifact.vmOptions.regions - images.keys
+        when {
+          images.isEmpty() -> {
+            log.info("No AMI found for {}", desiredAppVersion)
+            launchBake(artifact, desiredAppVersion)
           }
-        } else if (diff.hasChanges()) {
-          if (current == null) {
-            log.info("No AMI found for {}", artifact.name)
-          } else {
-            log.info("Image for {} delta: {}", artifact.name, diff.toDebug())
```

Comment:
Won't this logic trigger if ever there is a version that was baked with two different base images? Because you're only checking if the older one is populated, not if the older one is populated and the newer one isn't?

Reply:
I think you're right. Fixing

Reply:
Fixed